railgun/cmd/rootcmd/env.go

```diff
@@ -22,7 +29,10 @@ func bindEnvVars(cmd *cobra.Command, _ []string) {
 		}
 
 		if envValue, envSet := os.LookupEnv(envKey); envSet {
-			cmd.Flags().Set(f.Name, envValue)
+			if err := f.Value.Set(envValue); err != nil {
+				fmt.Fprintf(os.Stderr, "ERROR: could not set %s value from environment to %s: %s\n", envKey, envValue, err)
+				os.Exit(ExitCodeBadFlagValue)
```

Comment:
Suggest to return error and let the caller decide whether continue or exit from invalid setting.

The function is to trying to abstract the validation process, but it acts on behalf of caller - exit program, which is not the validation's job. a little bit conflict.

Reply:
Please return an error in 
https://github.com/Kong/kubernetes-ingress-controller/pull/1511/files#diff-4f4502a06cf38a24c5b34f902871497909c6c3794343ca49daeec5bc737d6ac2L16
for the failure in 
https://github.com/Kong/kubernetes-ingress-controller/pull/1511/files#diff-4f4502a06cf38a24c5b34f902871497909c6c3794343ca49daeec5bc737d6ac2R33
 and let the caller return from the program.

Reply:
I second this suggestion. :+1:

Reply:
Sounds good, 1455d231067c3007b6dd26b3563f70ec5238912d resolves this :+1: